### PR TITLE
Allow image affine to be calculated in advance for afq segmentation

### PR DESCRIPTION
**Pull request by 36000**

### CODE REVIEW

**Comment by 36000** [comment]
I believe the failure doesn't have to do with the changes I just made? Only failed in python 3.6

**Comment by arokem** [comment]
This looks good.  The Travis error was an OOM error, so hopefully restarting the build will be enough to resolve that.

File: AFQ/segmentation.py [hunk 1]
```diff
@@ -219,6 +222,7 @@ def _seg_afq(self, bundle_dict, streamlines, fdata=None, fbval=None,
         Quantification" PloS One 7 (11): e49790.
         """
         self.logger.info("Preparing Segmentation Parameters...")
+        self.img_affine = img_affine
```
Comment (arokem) [inline AFQ/segmentation.py]: Maybe we need to add some error handling: if users provide both `img_affine` and fdata, that could result in conflicting information.  So maybe users should  be allowed to provide one, or the other, but not both?

File: AFQ/segmentation.py [hunk 2]
```diff
@@ -218,7 +221,18 @@ def _seg_afq(self, bundle_dict, streamlines, fdata=None, fbval=None,
         "Tract Profiles of White Matter Properties: Automating Fiber-Tract
         Quantification" PloS One 7 (11): e49790.
         """
+        if img_affine is not None:
+            if (mapping is None or
+                fdata is not None or
+                fbval is not None or
+                    fbvec is not None):
```
Comment (arokem) [inline AFQ/segmentation.py]: I think that pep8 is upset about those `or`s being at the end of lines. See: https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

**Pull request merged**